sources/iTermApplicationDelegate.m

```diff
@@ -2063,7 +2063,17 @@ - (IBAction)loadRecording:(id)sender {
 #pragma mark - Private
 
 - (void)updateProcessType {
-    [[iTermApplication sharedApplication] setIsUIElement:[iTermPreferences boolForKey:kPreferenceKeyUIElement]];
+    bool enableLSUI = [iTermPreferences boolForKey:kPreferenceKeyUIElement];
+
+    if ([iTermAdvancedSettingsModel automaticallyToggleLSUI]) {
+        bool onlyHotKeyWindowsOpen = true;
+        for (PseudoTerminal* term in [self terminals]) {
```

Comment:
The `*` should touch `term`, not `PseudoTerminal`

Reply:
Thanks - I went with your functional suggestion below.